mcp/src/main/java/io/modelcontextprotocol/spec/McpSchema.java

```diff
@@ -964,61 +964,45 @@ public record CompleteCompletion(// @formatter:off
 			@JsonSubTypes.Type(value = EmbeddedResource.class, name = "resource") })
 	public sealed interface Content permits TextContent, ImageContent, EmbeddedResource {
 
-		String type();
+		default String type() {
```

Comment:
Since you are dealing with a sealed interface, you can use a [pattern switch](https://docs.oracle.com/en/java/javase/17/language/pattern-matching-switch.html) on `this` and have a complete coverage without the default case.

Reply:
pattern matching is available in Java 21+. The MCP project at the moment is configured for 17.

Reply:
Oh, it was a preview feature back then. My bad, sorry!